### PR TITLE
chore(percy): adding percy automated visual regression to web-components

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

Refs #2884 

### Description

This adds percy snapshot testing for web-components pull requests. The additional tasks have already been added to Jenkins.

### Changelog

**New**

- Percy automated visual regression testing for web components
- Removed redundant/out-of-date devDependencies

### CODE REVIEW

**Comment by jeffchew** [comment]
> @jeffchew Looks like it's complaining about missing `@carbon/ibmdotcom-styles` dependency
> image

Got it, checking it out now...

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 1]
```diff
@@ -96,7 +97,7 @@ export default {
     props: {
       'dds-masthead-container': {
         // Lets `<dds-masthead-container>` load the nav links if `CORS_PROXY` is set
-        navLinks: process.env.CORS_PROXY ? undefined : links,
+        navLinks: process.env.CORS_PROXY && !inPercy() ? undefined : links,
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:100]: Does this mean we don't want to use mock version of masthead links in Percy...?

Comment (jeffchew) [reply packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:100]: we actually do want to use the mocked version in percy. I think with this it means that we are indeed using the mocked version, `CORS_PROXY` is defined, and `inPercy()` returns true.

Comment (asudoh) [reply packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:100]: Got it, my bad... Thanks @jeffchew for clarifying!

**Pull request merged**